packages/eas-cli/src/commandUtils/context/contextUtils/loadServerSideEnvironmentVariablesAsync.ts

```diff
@@ -0,0 +1,58 @@
+import { ExpoGraphqlClient } from './createGraphqlClient';
+import { EnvironmentVariableEnvironment } from '../../../graphql/generated';
+import { EnvironmentVariablesQuery } from '../../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../../log';
+
+let cachedServerSideEnvironmentVariables: Record<string, string> | null = null;
+
+export async function loadServerSideEnvironmentVariablesAsync({
+  environment,
+  projectId,
+  graphqlClient,
+}: {
+  environment: EnvironmentVariableEnvironment;
+  projectId: string;
+  graphqlClient: ExpoGraphqlClient;
+}): Promise<Record<string, string>> {
+  // don't load environment variables if they were already loaded while executing a command
+  if (cachedServerSideEnvironmentVariables) {
```

Comment:
this should be memoized by environment instead of globally just in case a second call is added elsewhere during command execution that calls it with a different arg.

```
let cachedServerSideEnvironmentVariables: Map<
  EnvironmentVariableEnvironment,
  Record<string, string>
> = new Map();
```

```suggestion
  if (cachedServerSideEnvironmentVariables.has(environment)) {
```